Assets/Scripts/Actors/ActorComponents/PlayerActorInventory.cs

```diff
@@ -71,7 +71,30 @@ void CheckScroll()
 		}
 	}
 
-	public void CheckUseItem( RaycastHit hitInfo )
+	public bool CanUseItemWithoutTarget()
+	{
+		bool canUse = false;
+
+		if ( heldResources.Count > 0 )
+		{
+			if( !heldResources[resourceIndex].needsTarget )
+			{
+				canUse = true;
+			}
+		}
+
+		return canUse;
+	}
+
+	public void UseItem()
+	{
+		if ( !( heldResources[resourceIndex] is ResourceData ) )
+		{
+			SpawnBuddy();
+		}
+	}
+
+	public void UseItemWithTarget( RaycastHit hitInfo )
```

Comment:
These behaviours are all specific to Resources so we should probably rename all of them them UseResourceEtc